File: Chauffeur.Tests/Deliverables/QuitDeliverableTests.cs
using Chauffeur.Deliverables;
using System.Threading.Tasks;
using Xunit;

namespace Chauffeur.Tests.Deliverables
{
    public class QuitDeliverableTests
    {
        [Fact]
        public async Task ReturnsShutdownStatus()
        {
            var deliverable = new QuitDeliverable(null, new MockTextWriter());

            var response = await deliverable.Run("quit", new string[0]);

            Assert.Equal(DeliverableResponse.Shutdown, response);
        }
    }
}

File: src/Chauffeur.Tests/QuitDeliverableTests.fs
namespace Chauffeur.Tests

module QuitDeliverableTests =
    open Xunit
    open Chauffeur.Deliverables
    open FsUnit.Xunit
    open Chauffeur

    [<Fact>]
    let ``Shutdown response returned when run``() =
        let deliverable = QuitDeliverable(null, null)

        deliverable.Run "" Array.empty |> should equal DeliverableResponse.Shutdown